### PR TITLE
initial clic csr access test

### PR DESCRIPTION
**Pull request by silabs-hfegran**

### CODE REVIEW

**Comment by silabs-hfegran** [comment]
Note - as of now, this directed, self-checking test passes when run without ISS, currently fails with. Investigating why that happens

File: cv32e40s/tests/programs/custom/clic/clic.c [hunk 1]
```diff
@@ -0,0 +1,573 @@
+//
+// Copyright 2022 Silicon Labs, Inc.
```
Comment (silabs-robin) [inline cv32e40s/tests/programs/custom/clic/clic.c:2]: Does it not need some OpenHW name in the copyright list too?
(I have no idea what the rules are.)

Comment (silabs-hfegran) [reply cv32e40s/tests/programs/custom/clic/clic.c:2]: That is not the case for our rtl repositories - @MikeOpenHWGroup can you comment on this? 

As this is not derivative work, I do not really see why that is needed

Comment (MikeOpenHWGroup) [reply cv32e40s/tests/programs/custom/clic/clic.c:2]: The OpenHW Group does not need to hold the copyright on any code in our open source repositories, as long as the copyright holder assigns one of the premissive licenses we accept: EPL-2, Apache 2 or Solderpad 2.

We do have a lot of files in which multiple companies hold the copyright, but that is not required.  My guess is that either myself or SteveR created a header that attributed copyright to multiple organisations and it has been copied onto multiple files.

File: cv32e40s/tests/programs/custom/clic/clic.c [hunk 2]
```diff
@@ -0,0 +1,573 @@
...
+const uint32_t MSTATUS_MPIE_MASK = 0x1 << MSTATUS_MPIE_OFFSET;
+const uint32_t MCAUSE_MPP_MASK   = 0x3 << MCAUSE_MPP_OFFSET;
+const uint32_t MCAUSE_MPIE_MASK  = 0x1 << MCAUSE_MPIE_OFFSET;
+
+// ---------------------------------------------------------------
+// Test entry point
+// ---------------------------------------------------------------
+int main(int argc, char **argv){
+
+  uint32_t (*tests[NUM_TESTS])();
```
Comment (silabs-robin) [inline cv32e40s/tests/programs/custom/clic/clic.c:66]: Since it runs it is probably fine as it is, but doesn't it need a type for the argument?

**Review by silabs-robin — APPROVED** [review]
There is a clean pattern here and it seems to have been followed.
I haven't read everything 100%, but those parts that I read looked like they matched their purpose well.

File: cv32e40s/tests/programs/custom/clic/clic.c [hunk 3]
```diff
@@ -0,0 +1,1420 @@
...
+      mtvt_table_1023: .long . + 4
+      addi %[to_th], x0, 1
+      csrrs %[mepc], mepc, x0
+    )" : [to_th] "+r"(no_timeout_threshold), [mepc] "+r"(mepc_triggered)
+       :
+       : "memory"
+       );
+
+    cvprintf(V_DEBUG, "mepc: 0x%08lx\n", mepc_triggered);
+    // If this j is included above, the compiler
```
Comment (silabs-hfegran) [inline cv32e40s/tests/programs/custom/clic/clic.c:1175]: fixme: deprecated comment

File: cv32e40s/tests/programs/custom/clic/clic.c [hunk 4]
```diff
@@ -0,0 +1,1420 @@
...
+  mepc_triggered = 0;
+  vp_assert_irq(clic_vector, 10);
+
+  while (no_timeout_threshold) {
+    no_timeout_threshold--;
+  }
+  cvprintf(V_MEDIUM, "Expect non-zero mepc, MEPC = 0x%08lx\n", mepc_triggered);
+  if (ABORT_ON_ERROR_IMMEDIATE) assert(mepc_triggered != 0);
+
+  //max(SMCLIC_ID_WIDTH+2, 6)
```
Comment (silabs-hfegran) [inline cv32e40s/tests/programs/custom/clic/clic.c:1258]: remove comment

File: cv32e40s/tests/programs/custom/clic/clic.c [hunk 5]
```diff
@@ -0,0 +1,1420 @@
...
+        )"
+          : [cfg_vec] "+r"(pmpcfg_vector)
+          : [tmp] "r"(temp)
+          : "t0"
+          );
+    break;
+  }
+
+  __asm__ volatile ( R"( csrrs %[pmpvec], pmpcfg0, x0)" : [pmpvec] "+r"(pmpcfg_vector): :);
+  //__asm__ volatile ( R"(fence.i)":::);
```
Comment (silabs-hfegran) [inline cv32e40s/tests/programs/custom/clic/clic.c:568]: remove commented code

File: cv32e40s/tests/programs/custom/clic/clic.c [hunk 6]
```diff
@@ -0,0 +1,1420 @@
...
+      ((mseccfg.mml  << 0) & 0x1));
+
+  __asm__ volatile ( R"(
+    csrrs x0, mseccfg, %[cfg_vec]
+  )"
+      :
+      : [cfg_vec] "r"(mseccfg_vector)
+      :);
+
+  //__asm__ volatile ( R"(fence.i)":::);
```
Comment (silabs-hfegran) [inline cv32e40s/tests/programs/custom/clic/clic.c:372]: remove commented code

File: cv32e40s/tests/programs/custom/clic/clic.c [hunk 7]
```diff
@@ -0,0 +1,1420 @@
...
+const uint32_t MSTATUS_MPP_MASK  = 0x3 << MSTATUS_MPP_OFFSET;
+const uint32_t MSTATUS_MPIE_MASK = 0x1 << MSTATUS_MPIE_OFFSET;
+const uint32_t MCAUSE_MPP_MASK   = 0x3 << MCAUSE_MPP_OFFSET;
+const uint32_t MCAUSE_MPIE_MASK  = 0x1 << MCAUSE_MPIE_OFFSET;
+
+// Print verbosity, consider implementing this as a virtual
+// peripheral setting to be controlled from UVM.
+const verbosity_t global_verbosity = V_LOW;
+
+//volatile uint32_t (*mtvt_table_ptr);
```
Comment (silabs-hfegran) [inline cv32e40s/tests/programs/custom/clic/clic.c:129]: remove commented code

**Pull request merged**